### DISCUSSION TITLE
how to get timing path information

### DISCUSSION THREAD

**Discussion by msingh9** [The-OpenROAD-Project/OpenROAD#3029] [Q&A]
I am interested to find out my worst timing path. I can run following command to get path objects, but how do I get full report like start point, logic gates in between and end point.

openroad> find_timing_paths -from * -to * -group_count 10 -sort_by_slack
_d02fb74000000000_p_PathEnd _1005b54000000000_p_PathEnd _1040b44000000000_p_PathEnd _1006b54000000000_p_PathEnd _3071b64000000000_p_PathEnd _b0bfb54000000000_p_PathEnd _d0173d3f00000000_p_PathEnd _20cdb54000000000_p_PathEnd _9091bc4000000000_p_PathEnd _003db64000000000_p_PathEnd

**Comment by maliberty** [comment]
From the docs for find_timing_paths https://github.com/The-OpenROAD-Project/OpenSTA/blob/master/doc/OpenSTA.pdf

```
Use the get_property function to access properties of the paths.
```

**Comment by msingh9** [comment]
Thank you very much for prompt response. This works fantastically. I wrote a small TCL function to nicely print complete timing path with the arrival time. Maybe, it will be helpful to someone. 

```
### Document this ####
Usage: report_timing_paths <args>
Example: report_timing_paths -from * -to * -group_count 2 -sort_by_slack 

proc report_timing_paths {args} {
    set path_ends [sta::find_timing_paths_cmd "find_timing_paths" args]
    set pathno 1
    foreach path_end $path_ends {
        set start_clock [get_property [get_property $path_end startpoint_clock] full_name]
        set start_point [get_property [get_property $path_end startPoint] full_name]
        set end_point [get_property [get_property $path_end endPoint] full_name]
        set end_clock [get_property [get_property $path_end endpoint_clock] full_name]
        set slack [get_property $path_end slack]
        puts "Path ${pathno} : "
        puts " [format "%-80s" $start_point], ${start_clock}"
        set points [get_property $path_end points]
        foreach point $points {
            set arrival [get_property $point arrival]
            set pin [get_property [get_property $point pin] full_name]
            set cslack [get_property $point slack]
            puts " [format "%-80s" $pin], $arrival"
        }
        puts " [format "%-80s" $end_point], ${end_clock}, [format "%.3f" $slack]"
        incr pathno
        puts ""
    }
}
```

**Reply by daquintero** [reply]
It already is useful, cheers @msingh9
